numba/tests/test_dictobject.py

```diff
@@ -100,6 +100,19 @@ def foo(n, target, default):
         self.assertEqual(foo(5, 3, -1), 3)
         self.assertEqual(foo(5, 5, -1), -1)
 
+    def test_dict_get_with_non_none_default(self):
+        # Issue 5896
+        tup_type = types.Tuple.from_types([int32, int32])
+        @njit
+        def foo():
+            d = dictobject.new_dict(int32, tup_type)
+            d[1] = (2, 3)
+            valid = d.get(1, (4, 5))
+            invalid = d.get(0, (4, 5))
```

Comment:
I don't think this will suffice as a faithful reproduction of the original reproducer in #5896 since the final indexing into the returned (Optional) Tuple is missing.

If I run this PR on the following reproducer 

```
from numba import njit


@njit
def main():
    ds = {1: (2, 3)}       # (A)
    x = ds.get(1, (4, 5))  # (B)
    #x = ds[1]            # (B2)
    y = x[1]               # (C)
    return y


print(main())
```

I get the following, different, error:

```
Traceback (most recent call last):
  File "foo078.py", line 13, in <module>
    print(main())
  File "/Users/vhaenel/git/numba/numba/core/dispatcher.py", line 434, in _compile_for_args
    raise e
  File "/Users/vhaenel/git/numba/numba/core/dispatcher.py", line 367, in _compile_for_args
    return self.compile(tuple(argtypes))
  File "/Users/vhaenel/git/numba/numba/core/compiler_lock.py", line 32, in _acquire_compile_lock
    return func(*args, **kwargs)
  File "/Users/vhaenel/git/numba/numba/core/dispatcher.py", line 808, in compile
    cres = self._compiler.compile(args, return_type)
  File "/Users/vhaenel/git/numba/numba/core/dispatcher.py", line 78, in compile
    status, retval = self._compile_cached(args, return_type)
  File "/Users/vhaenel/git/numba/numba/core/dispatcher.py", line 92, in _compile_cached
    retval = self._compile_core(args, return_type)
  File "/Users/vhaenel/git/numba/numba/core/dispatcher.py", line 105, in _compile_core
    cres = compiler.compile_extra(self.targetdescr.typing_context,
  File "/Users/vhaenel/git/numba/numba/core/compiler.py", line 603, in compile_extra
    return pipeline.compile_extra(func)
  File "/Users/vhaenel/git/numba/numba/core/compiler.py", line 339, in compile_extra
    return self._compile_bytecode()
  File "/Users/vhaenel/git/numba/numba/core/compiler.py", line 401, in _compile_bytecode
    return self._compile_core()
  File "/Users/vhaenel/git/numba/numba/core/compiler.py", line 381, in _compile_core
    raise e
  File "/Users/vhaenel/git/numba/numba/core/compiler.py", line 372, in _compile_core
    pm.run(self.state)
  File "/Users/vhaenel/git/numba/numba/core/compiler_machinery.py", line 341, in run
    raise patched_exception
  File "/Users/vhaenel/git/numba/numba/core/compiler_machinery.py", line 332, in run
    self._runPass(idx, pass_inst, state)
  File "/Users/vhaenel/git/numba/numba/core/compiler_lock.py", line 32, in _acquire_compile_lock
    return func(*args, **kwargs)
  File "/Users/vhaenel/git/numba/numba/core/compiler_machinery.py", line 291, in _runPass
    mutated |= check(pss.run_pass, internal_state)
  File "/Users/vhaenel/git/numba/numba/core/compiler_machinery.py", line 264, in check
    mangled = func(compiler_state)
  File "/Users/vhaenel/git/numba/numba/core/typed_passes.py", line 442, in run_pass
    NativeLowering().run_pass(state)
  File "/Users/vhaenel/git/numba/numba/core/typed_passes.py", line 370, in run_pass
    lower.lower()
  File "/Users/vhaenel/git/numba/numba/core/lowering.py", line 217, in lower
    self.library.add_ir_module(self.module)
  File "/Users/vhaenel/git/numba/numba/core/codegen.py", line 206, in add_ir_module
    ll_module = ll.parse_assembly(ir)
  File "/Users/vhaenel/miniconda3/envs/numba_3.8/lib/python3.8/site-packages/llvmlite/binding/module.py", line 25, in parse_assembly
    raise RuntimeError("LLVM IR parsing error\n{0}".format(errmsg))
RuntimeError: Failed in nopython mode pipeline (step: nopython mode backend)
LLVM IR parsing error
<string>:336:3: error: instruction forward referenced with type 'i64'
  %".213" = extractvalue {[2 x i64], i1} %".204", 0
```